core/src/test/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotServiceSpec.scala

```diff
@@ -238,6 +238,48 @@ class SnapshotServiceSpec extends AnyWordSpecLike with Matchers with MockitoSuga
+      when(mockSamDAO.userHasAction(ArgumentMatchers.eq(SamResourceTypeNames.workspace), any[String], any[SamResourceAction], any[UserInfo])).thenReturn(Future.successful(true))
+      when(mockSamDAO.listPoliciesForResource(ArgumentMatchers.eq(SamResourceTypeNames.workspace), any[String], any[UserInfo])).thenReturn(Future.successful(Set(SamPolicyWithNameAndEmail(SamWorkspacePolicyNames.projectOwner, SamPolicy(Set(WorkbenchEmail(userInfo.userEmail.value)), Set.empty, Set.empty), WorkbenchEmail("")))))
+
+      val mockBigQueryServiceFactory = mock[GoogleBigQueryServiceFactory](RETURNS_SMART_NULLS)
+      when(mockBigQueryServiceFactory.getServiceFromCredentialPath(any[String], any[GoogleProject])).thenReturn(MockBigQueryServiceFactory.ioFactory().getServiceForPet("foo", GoogleProject("foo")))
+
+      val mockWorkspaceManagerDAO = mock[WorkspaceManagerDAO](RETURNS_SMART_NULLS)
+
+      val snapshotDataReferenceId = UUID.randomUUID()
+      when(mockWorkspaceManagerDAO.getDataReference(any[UUID], any[UUID], any[OAuth2BearerToken])).thenReturn(new DataReferenceDescription().referenceId(snapshotDataReferenceId).name("foo").description("").workspaceId(UUID.randomUUID()).cloningInstructions(CloningInstructionsEnum.NOTHING))
+
+      when(mockWorkspaceManagerDAO.getBigQueryDatasetReferenceByName(any[UUID], any[String], any[OAuth2BearerToken])).thenReturn(new BigQueryDatasetReference().metadata(new DataReferenceMetadata().referenceId(UUID.randomUUID())).dataset(new GoogleBigQueryDatasetUid))
+
+      val workspace = minimalTestData.workspace
+
+      val snapshotService = SnapshotService.constructor(
+        slickDataSource,
+        mockSamDAO,
+        mockWorkspaceManagerDAO,
+        mockBigQueryServiceFactory,
+        "fake-terra-data-repo-dev",
+        fakeCredentialPath,
+        fakeRawlsClientEmail,
+        fakeDeltaLayerStreamerEmail
+      )(userInfo)
+
+      val deltaLayerDatasetName = snapshotService.generateDatasetName(snapshotDataReferenceId)
+
+      val snapshotUUID = UUID.randomUUID()
+
+      Await.result(snapshotService.deleteSnapshot(workspace.toWorkspaceName, snapshotUUID.toString), Duration.Inf)
+
+      verify(mockWorkspaceManagerDAO, times(1)).getDataReference(ArgumentMatchers.eq(workspace.workspaceIdAsUUID), ArgumentMatchers.eq(snapshotUUID),any[OAuth2BearerToken])
+      verify(mockWorkspaceManagerDAO, times(1)).deleteDataReference(ArgumentMatchers.eq(workspace.workspaceIdAsUUID), ArgumentMatchers.eq(snapshotUUID), any[OAuth2BearerToken])
+      verify(mockWorkspaceManagerDAO, times(1)).getBigQueryDatasetReferenceByName(ArgumentMatchers.eq(workspace.workspaceIdAsUUID), ArgumentMatchers.eq(deltaLayerDatasetName), any[OAuth2BearerToken])
+      verify(mockWorkspaceManagerDAO, times(1)).deleteBigQueryDatasetReference(ArgumentMatchers.eq(workspace.workspaceIdAsUUID), any[UUID], any[OAuth2BearerToken])
+      verify(mockBigQueryServiceFactory, times(1)).getServiceFromCredentialPath(any[String], any[GoogleProject])
+
+    }
+
```

Comment:
when we add integration tests (AS-688), should probably add an integration test for the delete case too.